src/clj/harja/palvelin/raportointi/raportit/sanktiot.sql

```diff
@@ -37,7 +37,14 @@ WHERE ((:urakka::INTEGER IS NULL AND u.urakkanro IS NOT NULL) OR u.id = :urakka)
                                                                                         WHERE hallintayksikko =
                                                                                               :hallintayksikko) AND u.urakkanro IS NOT NULL))
       AND s.poistettu IS NOT TRUE
-      AND s.perintapvm BETWEEN :alku AND :loppu
+      -- jos hakurange sisältää urakan viimeisen kuukauden, mahdolliset urakan päättymisen jälkeen tulleet sanktiot sisällytetään siihen  
+      AND ((s.perintapvm BETWEEN :alku AND :loppu) OR
+          (CASE 
+                date_part('year', :loppu::date)::integer = date_part('year', u.loppupvm)::integer 
+                AND date_part('month', :loppu::date)::integer = date_part('month', u.loppupvm)::integer
+           WHEN TRUE THEN s.perintapvm > u.loppupvm 
```

Comment:
Mulle ei ole tuttu tällainen rakenne, että CASE:n jälkeen ei tule WHEN vaan ehto. Eli tämäkin varmaan toimii, mutta onko olemassa jokin syy, miksi CASE ei olisi tämän muotoinen:
(CASE 
WHEN
       date_part('year', :loppu::date)::integer = date_part('year', u.loppupvm)::integer 
                AND date_part('month', :loppu::date)::integer = date_part('month', u.loppupvm)::integer 
AND s.perintapvm > u.loppupvm
          THEN TRUE
ELSE FALSE
           END)

Reply:
joo, tää on mullekin uusi rakenne, mitä ajattelin kokeilla. Bongasin sen tuolta postgresqln dokkareista, jossa mainittiin, että se toimii switch casen tapaan mikä tässä vaikutti ihan näppärältä, kun kiinnostaa vain tuo yksi tapaus tuossa ja sen tulos. Ei sinällään muuta syytä sen käyttöön.